File: testng-core/src/test/java/org/testng/util/StringsTest.java
package org.testng.util;

import static org.testng.Assert.assertEquals;

import org.testng.annotations.Test;

/** Unit tests for {@link Strings} */
public class StringsTest {
  @Test
  public void joinEmptyArray() {
    String[] emptyArray = new String[0];
    assertEquals(Strings.join(",", emptyArray), "");
  }

  @Test
  public void joinArrayWithOneElement() {
    String[] array = new String[] {"one"};
    assertEquals(Strings.join(",", array), "one");
  }

  @Test
  public void joinArrayWithTwoElements() {
    String[] array = new String[] {"one", "two"};
    assertEquals(Strings.join(",", array), "one,two");
  }
}

File: testng-core/src/test/kotlin/org/testng/util/StringsTest.kt
package org.testng.util

import org.assertj.core.api.Assertions.assertThat
import org.testng.annotations.Test

class StringsTest {
    @Test
    fun testValueOf() {
        val input = mapOf(
            Pair("skill", "Kung-Fu"),
            Pair("expertise-level", "Master"),
            Pair("name", "Po")
        )
        val expected = "Kung-Fu Master Po"
        val actual = Strings.valueOf(input)
        assertThat(actual).isEqualTo(expected)
    }

    @Test
    fun testEscapeHtml() {
        val input = "&<>"
        val expected = "&amp;&lt;&gt;"
        val actual = Strings.escapeHtml(input)
        assertThat(actual).isEqualTo(expected)
    }
}
